src/Text.Analyzers/Core/IdentifiersShouldBeSpelledCorrectly.cs

```diff
@@ -255,21 +263,39 @@ private static void OnCompilationStart(CompilationStartAnalysisContext compilati
                 SymbolKind.Field,
                 SymbolKind.Parameter);
 
-            IEnumerable<CodeAnalysisDictionary> ReadDictionaries()
+            ImmutableArray<CodeAnalysisDictionary> ReadDictionaries()
             {
-                var fileProvider = AdditionalFileProvider.FromOptions(compilationStartContext.Options);
+                var fileProvider = AdditionalFileProvider.FromOptions(context.Options);
                 return fileProvider.GetMatchingFiles(@"(?:dictionary|custom).*?\.(?:xml|dic)$")
-                    .Select(CreateDictionaryFromAdditionalText)
+                    .Select(GetOrCreateDictionaryFromAdditionalText)
                     .Where(x => x != null)
-                    .ToList();
+                    .ToImmutableArray();
+
+                CodeAnalysisDictionary GetOrCreateDictionaryFromAdditionalText(AdditionalText additionalText)
+                {
+                    if (!_textToDictionary.TryGetValue(additionalText, out var dictionary))
+                    {
+                        dictionary = CreateAndAddDictionary(additionalText);
+                    }
+
+                    return dictionary;
+
+                    // Local function to avoid unnecessary lambda alloc in mainline case.
```

Comment:
❔ Wouldn't it only need to alloc the lambda if the lambda references `additionalText` from the containing scope? In this case, it only seems to reference `dictionary` from the current scope.

Reply:
I'll check the IL, but i believe it will still create a closure.